joblib/pool.py

```diff
@@ -549,7 +549,16 @@ def __init__(self, processes=None, temp_folder=None, max_nbytes=1e6,
         # self to ensure that this callback won't prevent garbage collection of
         # the pool instance and related file handler resources such as POSIX
         # semaphores and pipes
-        atexit.register(lambda: delete_folder(pool_folder))
+        def _cleanup():
+            # In some cases the Python runtime seems to set delete_folder to
+            # None just before exiting when accessing the delete_folder
+            # function from the closure namespace. So instead we reimport
+            # the delete_folder function explicitly.
+            # https://github.com/joblib/joblib/issues/328
+            from .pool import delete_folder
```

Comment:
Yes but no: we need to preserve the ability of joblib to be "vendored" inside other libraries like scikit-learn by only using relative imports.

I will try something else.